products/messaging/frontend/Campaigns/CampaignScene.tsx

```diff
@@ -19,10 +25,59 @@ export const scene: SceneExport = {
 
 export function CampaignScene(props: CampaignSceneLogicProps = {}): JSX.Element {
     const { currentTab } = useValues(campaignSceneLogic)
+    const [isStopModalOpen, setIsStopModalOpen] = useState(false)
+    const [stopType, setStopType] = useState<'trigger' | 'all'>('trigger')
+    const { user } = useValues(userLogic)
+    const { currentTeam } = useValues(teamLogic)
 
     const logic = campaignLogic(props)
-    const { campaignChanged, originalCampaign, isCampaignSubmitting } = useValues(logic)
-    const { submitCampaign, resetCampaign } = useActions(logic)
+    const { campaignChanged, originalCampaign, isCampaignSubmitting, campaign } = useValues(logic)
+    const { submitCampaign, resetCampaign, saveCampaign } = useActions(logic)
+
+    useEffect(() => {
+        if (!currentTeam?.live_events_token) {
+            return
+        }
+
+        const url = new URL(`${liveEventsHostOrigin()}/events`)
+        url.searchParams.append('eventType', 'campaign_view')
+        url.searchParams.append('distinctId', user?.email || '')
+        url.searchParams.append('token', currentTeam.live_events_token)
+
+        const eventSource = new EventSource(url.toString())
+
+        eventSource.onmessage = (event) => {
+            const data = JSON.parse(event.data)
+            if (data.event === 'campaign_view' && data.properties?.campaign_id === props.id) {
+                // We're not displaying viewers anymore, so we can ignore this
+            }
+        }
+
+        // Send initial view event
+        void fetch(`${liveEventsHostOrigin()}/stats`, {
+            headers: {
+                Authorization: `Bearer ${currentTeam.live_events_token}`,
+            },
+            method: 'POST',
+            body: JSON.stringify({
+                event: 'campaign_view',
+                properties: {
+                    campaign_id: props.id,
+                    user_email: user?.email,
+                },
+            }),
+        })
+
+        return () => {
+            eventSource.close()
+        }
+    }, [props.id, currentTeam?.live_events_token, user?.email])
```

Comment:
Genuinely don't get what this is about tbh?

Reply:
On second read - are you just trying to track an event?
```suggestion
    useEffect(() => {
        posthog.capture('campaign_view', { campaign_id: props.id })
    }, [props.id])
```